BLSQ/iaso
# IA-2292 500 when using a csv with semicolon in batch user creation + add projects and user roles

Explain what problem this PR is resolving

Related JIRA tickets : IA-2292 and IA-2289

## Self proofreading checklist



## Changes

Field Updates : 

Mandatory columns  have been updated to make it less error prone. 

New column " user_roles " and " projects " have been added.

Mandatory columns are : ["username", "password", "email", "first_name", "last_name", "orgunit", "profile_language", "dhis2_id", "permissions", "user_roles", "projects"]

I added a check of csv delimiter to handle files separated by both coma and semicolon. For some reason sometimes ne Sniffer doesnt work and result into a error. So the code check first with Sniffer, if it fails it try a " manual " check by parsing the file. 

Also, more tests.

## How to test

Explain how to test your PR.
Test with a CSV created with Excel.
Upload a CSV file with semicolon separator, it should work. Also test the new fields. Use one of the working templates to test.
